modules/k8s/namespace.go

```diff
@@ -17,15 +17,23 @@ func CreateNamespace(t testing.TestingT, options *KubectlOptions, namespaceName
 
 // CreateNamespaceE will create a new Kubernetes namespace on the cluster targeted by the provided options.
 func CreateNamespaceE(t testing.TestingT, options *KubectlOptions, namespaceName string) error {
+	namespaceObject := metav1.ObjectMeta{
+		Name: namespaceName,
+	}
+	err := CreateNamespaceWithMetadataE(t, options, namespaceObject)
+	return err
+}
+
+// CreateNamespaceWithMetadataE will create a new Kubernetes namespace on the cluster targeted by the provided options and
+// expects the entire namespace ObjectMeta to be passed in.
+func CreateNamespaceWithMetadataE(t testing.TestingT, options *KubectlOptions, namespaceObjectMeta metav1.ObjectMeta) error {
```

Comment:
NIT: could you also add a `CreateNamespaceWithMetadata` version of this function?

Reply:
done! let me know if that's what you had in mind. I also added a test to cover it